app/test_main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+from typing import Any, Type
+from unittest.mock import patch
+
+import pytest
+
+from app.main import can_access_google_page
+from pytest import fixture
+
+
+@fixture
+def manager_fixture() -> Any:
+    with (patch("app.main.valid_google_url") as valid_url,
+          patch("app.main.has_internet_connection") as has_connection):
+        yield valid_url, has_connection
+
+
+@pytest.mark.parametrize(
+    "url, connection, result",
+    [
+        (True, True, "Accessible"),
+        (False, True, "Not accessible"),
+        (True, False, "Not accessible"),
+        (False, False, "Not accessible")
+    ]
+)
+def test_can_access_google_page(
```

Comment:
You can just use mock.patch decorators to the tast instead of writing manager_fixture function with context manager
It will give the same affect and simplify the code